src/core/CLucene/util/CLStreams.h

```diff
@@ -207,7 +207,8 @@ class CLUCENE_EXPORT SStringReader : public Reader {
                 tmp = (T *) realloc(tmp, sizeof(T) * (length + 1));
                 this->buffer_size = length;
             }
-            memcpy(tmp, _value, length + 1);
+            memcpy(tmp, _value, length);
```

Comment:
memcpy(tmp, _value, length * sizeof(T));

Reply:
I think we should not do use length * sizeof(T), since the input size is just length. I'll change it to use different way for sizeof(T) == 1 and others.